lib/WebDriver/ServiceFactory.php

```diff
@@ -103,6 +103,11 @@ public function setServiceClass($serviceName, $className)
             $className = '\\' . $className;
         }
 
+        // Flush outdated service cache
+        if (isset($this->serviceClasses[$serviceName]) && $this->serviceClasses[$serviceName] !== $className) {
+            unset($this->services[$serviceName]);
```

Comment:
It's not PHP version specific; ServiceFactory::getService() would ignore any updated serviceClass if an outdated service exists for any prior value. This is necessary to force the `new $className` to trigger in any subsequent getService() call.

I would replace with setService() to allow a mock to be directly injected.